### PR TITLE
Allow more flexible processing of return groups in ground filters

### PR DESCRIPTION
**Pull request by chambbj**
Continue to default to processing last and only returns, but allow users to
specify any combination of "last", "first", "intermediate", and "only".

### CODE REVIEW

File: filters/PMFFilter.cpp [hunk 1]
```diff
@@ -88,14 +85,15 @@ void PMFFilter::addArgs(ProgramArgs& args)
...
-        m_args->m_exponential, true);
+             m_args->m_exponential, true);
     args.add("ignore", "Ignore values", m_args->m_ignored);
-    args.add("initial_distance", "Initial distance",
-        m_args->m_initialDistance, 0.15);
-    args.add("last", "Consider last returns only?", m_args->m_lastOnly, true);
+    args.add("initial_distance", "Initial distance", m_args->m_initialDistance,
+             0.15);
+    args.add("returns", "Include only returns?", m_args->m_returns,
+             {"last", "only"});
```
Comment (abellgithub) [inline filters/PMFFilter.cpp]: It seems best to either validate this in initialize() or convert automatically using something like ReturnType, where you define ReturnType to encode the possible values.  There isn't a great way to do the second option right now with ProgramArgs, but I'm in the process of adding something.

File: pdal/Segmentation.cpp [hunk 2]
```diff
@@ -141,5 +141,67 @@ void segmentLastReturns(PointViewPtr input, PointViewPtr last,
...
+        {
+            if (outputTypes & returnOnly)
+                first->appendPoint(*input.get(), i);
+            else
+                second->appendPoint(*input.get(), i);
+        }
+        else
+            second->appendPoint(*input.get(), i);
+    }
+    std::cerr << first->size() << ", " << second->size() << std::endl;
```
[the remark this answers is not in the document]
Comment (chambbj) [reply pdal/Segmentation.cpp]: D'oh. Will remove.

File: pdal/Segmentation.cpp [hunk 3]
```diff
@@ -141,5 +141,67 @@ void segmentLastReturns(PointViewPtr input, PointViewPtr last,
...
+        else if (r == "only")
+            outputTypes |= returnOnly;
+    }
+
+    for (PointId i = 0; i < input->size(); ++i)
+    {
+        PointRef p = input->point(i);
+        uint8_t rn = input->getFieldAs<uint8_t>(Dimension::Id::ReturnNumber, i);
+        uint8_t nr = input->getFieldAs<uint8_t>(Dimension::Id::NumberOfReturns, i);
+        if ((rn == 1) && (nr > 1))
```
[the remark this answers is not in the document]
Comment (chambbj) [reply pdal/Segmentation.cpp]: Sure, it could be simplified some.

File: filters/PMFFilter.cpp (hunk 1 above)
Comment (abellgithub) [reply filters/PMFFilter.cpp]: I'm going to test after lunch.

**Comment by chambbj** [comment]
See if my latest commit answers the mail @abellgithub. I decided just to leave it in `prepared` rather than introducing `initialize`. If there is a compelling reason to do it in `initialize`, just let me know.

**Pull request merged**